aiida/orm/nodes/process/process.py

```diff
@@ -479,7 +479,23 @@ def is_valid_cache(self):
 
         :returns: True if this process node is valid to be used for caching, False otherwise
         """
-        return super().is_valid_cache and self.is_finished
+        if not (super().is_valid_cache and self.is_finished):
+            return False
+        try:
+            process_class = self.process_class
+        except Exception as exc:  # pylint: disable=broad-except
```

Comment:
Should we not make this a `ValueError`?

Reply:
This is the case where the process class might not be loadable (as discussed above). I don't think we know all possible causes for this, do we?

Reply:
But the `process_class` property already catches all these different potential reasons and converts them in a `ValueError`. Of course `process_class` implementation may have missed one but then it is a bug and I'd rather have it shown and fixed then hidden by broad except.

Reply:
Never mind, you are right.